agent/agent.go

```diff
@@ -648,6 +635,19 @@ func (a *Agent) wait(ctx, taskCtx context.Context, tc *taskContext, heartbeat ch
 			tc.logger.Execution().Debugf("Found no OOM kill (in %.3f seconds).", time.Since(startTime).Seconds())
```

Comment:
These lines still have to obey the order of operations that they had before to avoid changing the order that timeout/post/OOM kill checking run, so the timeout and OOM tracker handling have to run after `endTaskResponse` and before the post block runs.

Reply:
endTaskResponse runs in  a.finishTask on line 609, so this runs after that. This whole thing runs in runTask, which runs before runPostGroupCommands in the loop.

Reply:
This is talking about the order of [`runPostTaskCommands`](https://github.com/evergreen-ci/evergreen/blob/main/agent/agent.go#L680-L684) (post and teardown task) and `runEndTaskSync` with respect to these lines. Before this PR, these lines ran before `runPostTaskCommands`, but now they run after.